### PR TITLE
Bloom tokenizer

### PR DESCRIPTION
**Pull request by abuelnasr0**
The conversion script worked alright and the backbone and tokenizer generated the same output as hugging face.
here is a gist: https://colab.research.google.com/gist/abuelnasr0/a603fd2077b6cc579b51876b8cefecd9/bloom.ipynb

I have also changed the preset names. @mattdangerw please take a look at them and tell me If you want to change them.

regarding uploading the model variations into kaggle. I have an issue. first of all I can't upload it from my local machine because I have bad internet connection. and we have limited internet. my internet quota will end If I tried to upload or download models. I only work at colab or kaggle. 
so I tried to put a link to a google drive folder containing the preset model and it didn't work.
I also tried to add variation from a notebook output. but kaggle doesn't show the notebook that outputs the preset as option to me, although it's public and has an Output Data. the notebook link: https://www.kaggle.com/code/mohamedabuelnasr/bloom-model-saving

any suggestions to upload the preset ?

**UPDATE**
I managed to upload the preset correctly after zipping the "assets" directory in the Kaggle output and copying it's link and other files link from kaggle notebook output.
But it would be useful if someone could tell me why my notebook output doesn't appear in the "Notebook output" section while uploading the model.

### CODE REVIEW

**Comment by abuelnasr0** [comment]
I managed to upload the preset into kaggle by copying each file link from kaggle notebook output. but I can't put the "merges.txt" and "vocabulary.json" files into "assets/tokenizer" directory

**Comment by abuelnasr0** [comment]
The Preset test will fail until I have permission to make the model public at kaggle. I tried loading the preset in a kaggle notebook because I have access to the link and it worked fine.

**Comment by mattdangerw** [comment]
@abuelnasr0 can you share it with me? https://www.kaggle.com/matthewdwatson

Public sharing is not rolled out to Kaggle quite yet.

File: tools/checkpoint_conversion/convert_bloom_checkpoints.py [hunk 1]
```diff
@@ -12,51 +12,86 @@
...
-from keras_nlp.models.bloom.bloom_backbone import BloomBackbone
+os.environ["KERAS_BACKEND"] = "torch"
+os.environ["CUDA_VISIBLE_DEVICES"] = "-1"
+
+import numpy as np  # noqa: E402
+import torch  # noqa: E402
+import transformers  # noqa: E402
+from absl import app  # noqa: E402
+from absl import flags  # noqa: E402
+from huggingface_hub import snapshot_download  # noqa: E402
```
Comment (mattdangerw) [inline tools/checkpoint_conversion/convert_bloom_checkpoints.py:26]: nit: just `import huggingface_hub` here (makes it easier to track what is coming from where)

File: tools/checkpoint_conversion/convert_bloom_checkpoints.py [hunk 2]
```diff
@@ -141,54 +176,74 @@ def convert_checkpoints(hf_model):
...
-    keras_model.save_weights(f"{FLAGS.preset}.weights.h5")
 
-    return keras_model
+def validate_output(
+    hf_model,
+    keras_model,
+    hf_tokenizer,
+    keras_tokenizer,
+):
+    print("⏺ checking Tokenizer and Backbone outputs")
```
Comment (mattdangerw) [inline tools/checkpoint_conversion/convert_bloom_checkpoints.py:186]: We can probably ditch this print, we are already printing at the end of each "stage"

**Review by mattdangerw — APPROVED** [review]
Really awesome work! If you can share the one conversion you did with me. I will move it over to the Keras team.

We can probably just go with the one preset as we finish the model code. E.g. add the `CausalLM`. Once we have a full end to end demo with 560m, we can upload the rest.

**Comment by abuelnasr0** [comment]
@mattdangerw I have shared the model with you now.

**Comment by mattdangerw** [comment]
@abuelnasr0 Thanks!

I think this is already to go. Just copied the model you shared over to the Keras team. If tests look green, I will merge this in!

**Comment by mattdangerw** [comment]
Huh, need to figure out how to make the Keras model public. Will check on this tomorrow!

**Comment by mattdangerw** [comment]
Finally pulling this in! Sorry about the delays.

Can you add some generative stuff on top? E.g. the CausalLM class and co?

**Comment by abuelnasr0** [comment]
@mattdangerw of course. I will start by adding preprocessor then casual_lm_preprocessor and casual_lm.

**Pull request merged**